src/sentry/tasks/app_store_connect.py

```diff
@@ -53,6 +54,10 @@ def inner_dsym_download(project_id: int, config_id: str) -> None:
             if not build_state.fetched:
                 builds.append((build, build_state))
 
+    build_refresh_dates = project.get_option(appconnect.SYMBOL_SOURCE_REFRESH_PROP_NAME, default={})
+    build_refresh_dates[config_id] = datetime.now()
+    project.update_option(appconnect.SYMBOL_SOURCE_REFRESH_PROP_NAME, build_refresh_dates)
```

Comment:
This directly inserts an object in as a project option whereas a different option with a similarly complex value, symbol sources (`sentry:symbol_sources`) first converts its value to JSON before being inserted. 

Could there be a reason why the symbol sources uses JSON as opposed to inserting the list directly in to the project option, and should this also be serialized into a JSON blob?

@untitaker @jan-auer @mitsuhiko

Reply:
it doesn't matter IMO as long as the values are JSON-serializable

datetime I would convert into a primitive type

I hope that at some point we manage to change projectoptions to get rid of pickle at which point we'd not be able to store arbitrary objects like that

Reply:
cheeky question: would serializing the object before storing it in the option count as converting datetime to a primitive type?

Reply:
yes